nixos/modules/services/games/vrising-server.nix

```
@@ -0,0 +1,174 @@
+            CastleDamageMode = "Never";
+            BloodBoundEquipment = true;
+            CanLootEnemyContainers = false;
+            Death_DurabilityFactorLoss = 0.0;
+            Death_DurabilityLossFactorAsResources = 0.0;
+            AnnounceSiegeWeaponSpawn = false;
+          }'';
+        description = ''
+          Game settings as defined in <link xlink:href="https://cdn.stunlock.com/blog/2022/05/25083113/Game-Server-Settings.pdf">this file</link>.
+          Look at the reference to see what you can configure in this attribute set.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    warnings = (if cfg.hostSettings ? Password then [''
+      You have defined a host password using the hostSettings option, which
+      will be ignored. Use the dedicated passwordFile option instead.
+    ''] else
+      [ ]);
+
+    networking.firewall.allowedUDPPorts = mkIf cfg.openFirewall [
+      cfg.hostSettings.port
+      cfg.hostSettings.queryPort
+    ];
+
+    systemd.services.vrising-server = {
+      description = "V Rising dedicated server";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+
+      environment = {
+        # steamcmd uses $HOME/.local for temporary files
+        HOME = varLibStateDir;
+        saveDir = "./save-data";
+        hostSettings = builtins.toJSON cfg.hostSettings;
+        gameSettings = builtins.toJSON cfg.gameSettings;
+      };
+
+      serviceConfig = {
+        ExecStartPre = ''
+          ${pkgs.steamcmd}/bin/steamcmd \
+          +force_install_dir ${varLibStateDir} \
+          +login anonymous \
+          +app_update 1829350 \
+          +quit
+        '';
+        # sometimes winedevice processes are left running https://github.com/lutris/lutris/issues/4046
+        ExecStop = "${pkgs.wine64}/bin/wineserver -k";
+        WorkingDirectory = varLibStateDir;
+        Restart = "no";
+        StateDirectory = cfg.stateDir;
+        DynamicUser = true;
+        # in case execstop does not kill all winedevice processes
+        TimeoutStopSec = 10;
+        LoadCredential = "password:${cfg.passwordFile}";
+
+        # Hardening
+        SystemCallFilter = "@system-service @mount @debug";
+        SystemCallErrorNumber = "EPERM";
+        CapabilityBoundingSet = "";
+        RestrictNamespaces = "cgroup mnt user uts";
+        RestrictAddressFamilies = "AF_UNIX AF_INET AF_INET6 AF_NETLINK";
+      };
+
+      # script is needed here since loadcredential is not accessible in
+      # ExecPreStart. Wine sometimes fails with dll import errors, removing
+      # .wine resolves this, same issue as reported in the first comment of this
+      # PR https://github.com/NixOS/nixpkgs/pull/157921
+      script = ''
+        rm -rf .wine
+        mkdir -p $saveDir/Settings
+        ${pkgs.jq}/bin/jq -n --arg pw "$(< $CREDENTIALS_DIRECTORY/password )" \
+          '$ENV.hostSettings | fromjson | .Password = $pw' > $saveDir/Settings/ServerHostSettings.json
```

that'll leak the password in the command line of the jq call. better use `--slurpfile`, eg (untested!)
```suggestion
        ${pkgs.jq}/bin/jq -n --slurpfile pw <(jq -R . < "$CREDENTIALS_DIRECTORY/password") \
          '.Password = $pw[0]' < ${hostSettings} > $saveDir/Settings/ServerHostSettings.json
```